airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/ReplicationView.tsx

```diff
@@ -68,14 +69,16 @@ export const ReplicationView: React.FC<ReplicationViewProps> = ({ onAfterSaveSch
 
   const onSubmit = async (values: ValuesProps, formikHelpers?: FormikHelpers<ValuesProps>) => {
     const initialSyncSchema = connection?.syncCatalog;
+    const connectionAsUpdate = connection ? toWebBackendConnectionUpdate(connection) : undefined;
 
     await updateConnection({
+      ...connectionAsUpdate,
       ...values,
       connectionId,
+      // Use the name and status from the initial connection because these can be updating while refreshing the schema.
```

Comment:
Hmm, not sure if I'm following fully, does this mean it's possible that somebody would try to edit the name, but then it gets reverted once this request goes through?

Reply:
Right to explain the use case:
1. User presses refresh source schema
2. User updates name or disables/enables the connection
3. User updates the source settings and saves changes

Because both updating the name and updating the connection status can be done in between updating the schema, the connection object will be updated but not the connection object that contains the refreshed schema information.

Reply:
Gotcha, I think that makes sense to me, thanks for the explanation!